datastore/elastic_search/query.py

```diff
@@ -312,6 +314,14 @@ def full_text_query(connection, index_name, doc_type, entity_name, sentences, fu
     try:
         response = _run_es_search(connection, msearch=True, **kwargs)
         results = _parse_es_search_results(response.get("responses"))
+    except es_exceptions.NotFoundError as e:
+        raise DataStoreRequestException(f'NotFoundError in datastore query on index: {index_name}',
+                                        engine='elasticsearch', request=json.dumps(data),
+                                        response=json.dumps(response)) from e
+    except es_exceptions.ConnectionError as e:
+        raise DataStoreRequestException(f'Error while connecting to ES, in datastore query on index: {index_name}',
```

Comment:
I think we should raise a connection error itself to indicate that this is an infrastructure issue and not a ML code issue. That way we know when to ping devops and when to look at it ourselves.